pkg/scheduler/partition.go

```diff
@@ -1207,6 +1207,8 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
 
 // removeAllocation removes the referenced allocation(s) from the applications and nodes
 // NOTE: this is a lock free call. It must NOT be called holding the PartitionContext lock.
+//
+//nolint:funlen
 func (pc *PartitionContext) removeAllocation(release *si.AllocationRelease) ([]*objects.Allocation, *objects.Allocation) {
```

Comment:
It seems we can extract the code of generating `released` into a separate method. WDYT?